tests/sentry/integrations/utils/test_code_mapping.py

```diff
@@ -322,3 +327,79 @@ def test_normalized_stack_and_source_roots_starts_with_app_dot_dot_slash(self):
         )
         assert stacktrace_root == "app:///../"
         assert source_path == ""
+
+
+class TestGetSortedCodeMappingConfigs(TestCase):
+    def setUp(self):
+        super()
+        with assume_test_silo_mode(SiloMode.CONTROL):
+            self.integration = Integration.objects.create(provider="example", name="Example")
+            self.integration.add_organization(self.organization, self.user)
+            self.oi = OrganizationIntegration.objects.get(integration_id=self.integration.id)
+
+        self.repo = self.create_repo(
+            project=self.project,
+            name="getsentry/sentry",
+        )
+        self.repo.integration_id = self.integration.id
+        self.repo.provider = "example"
+        self.repo.save()
+
+    def test_get_sorted_code_mapping_configs(self):
+        # Created by the user, not well defined stack root
+        code_mapping1 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="",
+            source_root="",
+            automatically_generated=False,
+        )
+        # Created by automation, not as well defined stack root
+        code_mapping2 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="usr/src/getsentry/src/",
+            source_root="",
+            automatically_generated=True,
+        )
+        # Created by the user, well defined stack root
+        code_mapping3 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="usr/src/getsentry/",
+            source_root="",
+            automatically_generated=False,
+        )
+        # Created by the user, not as well defined stack root
+        code_mapping4 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="usr/src/",
+            source_root="",
+            automatically_generated=False,
+        )
+        # Created by automation, well defined stack root
+        code_mapping5 = self.create_code_mapping(
+            organization_integration=self.oi,
+            project=self.project,
+            repo=self.repo,
+            stack_root="usr/src/getsentry/src/sentry/",
+            source_root="",
+            automatically_generated=True,
+        )
+
+        # Expected configs: stack_root, automatically_generated
+        expected_config_order = [
```

Comment:
So user generated code mappings are prioritized over automatically generated ones?

Reply:
Yes that's a good callout. That's another thing that the sorting function does. It prioritizes first user generated over automatic mappings, and secondarily prioritizes stack roots that are more defined.